src/main/java/de/tum/in/www1/artemis/service/ParticipantScoreService.java

```diff
@@ -117,55 +119,47 @@ private List<ScoreDTO> calculateScores(Set<Exercise> exercises, Set<User> users,
             return List.of();
         }
 
-        Set<Exercise> individualExercises = exercises.stream().filter(exercise -> !exercise.isTeamMode()).collect(Collectors.toSet());
+        Set<Exercise> individualExercises = exercises.stream().filter(Predicate.not(Exercise::isTeamMode)).collect(Collectors.toSet());
         Set<Exercise> teamExercises = exercises.stream().filter(Exercise::isTeamMode).collect(Collectors.toSet());
 
         Course course = exercises.stream().findAny().orElseThrow(() -> new EntityNotFoundException("The result you are referring to does not exist"))
                 .getCourseViaExerciseGroupOrCourseMember();
 
-        // For every student we want to calculate the score
-        Map<Long, ScoreDTO> userIdToScores = users.stream().collect(Collectors.toMap(User::getId, user -> new ScoreDTO(user.getId(), user.getLogin(), 0.0, 0.0, 0.0)));
-
         // individual exercises
         // [0] -> User
         // [1] -> sum of achieved points in exercises
         List<Object[]> studentAndAchievedPoints = studentScoreRepository.getAchievedPointsOfStudents(individualExercises);
-        for (Object[] rawData : studentAndAchievedPoints) {
-            User user = (User) rawData[0];
-            double achievedPoints = rawData[1] != null ? ((Number) rawData[1]).doubleValue() : 0.0;
-            if (userIdToScores.containsKey(user.getId())) {
-                userIdToScores.get(user.getId()).pointsAchieved += achievedPoints;
-            }
-        }
+        Map<Long, Double> pointsAchieved = studentAndAchievedPoints.stream().collect(Collectors.toMap(rawData -> ((User) ((Object[]) rawData)[0]).getId(),
+                rawData -> ((Object[]) rawData)[1] != null ? ((Number) ((Object[]) rawData)[1]).doubleValue() : 0D));
```

Comment:
Not sure if this really is an improvement, the code is basically unreadable now.

Reply:
Please elaborate on which of the concepts are too advanced and might be unclear. 
I'm open to improvements/suggestions.

Reply:
@Strohgelaender @tobias-lippert I've refactored the code. Please let me know if this is now easier to understand.
IMO this is now even easier to read than the previous approach to construct this map.